src/cogent3/core/alignment.py

```diff
@@ -4158,17 +4162,17 @@ def get_projected_annotations(self, seq_name, *args):
         aln_annots = self.get_annotations_matching(*args)
         return [self.project_annotation(seq_name, a) for a in aln_annots]
 
-    def get_annotations_from_seq(self, seq_name, annotation_type="*", *args):
+    def get_annotations_from_seq(self, seq_name, annotation_type="*", name=None, *args):
         aligned = self.named_seqs[seq_name]
         return aligned.get_annotations_matching(
-            self, annotation_type=annotation_type, *args
+            self, annotation_type=annotation_type, name=name, *args
         )
 
-    def get_annotations_from_any_seq(self, annotation_type="*", *args):
+    def get_annotations_from_any_seq(self, annotation_type="*", name=None, *args):
```

Comment:
We need to support the extended querying (e.g. `extend_query=True`) so add `**kwargs` to this method and the called method.